spec/system/demo_splash_spec.rb

```diff
@@ -162,19 +162,26 @@
 
         fill_in 'Search...', with: 'A Second Persona'
         expect(page).to have_text('A Second Persona')
-        expect(page).to have_no_text('The Everyperson')
+        before = page.text
+        sleep 1
+        after = page.text
+        if before != after
+          puts "Before: #{before.inspect}\nAfter: #{after.inspect}"
+        end
+
+        expect(page).not_to have_text('The Everyperson')
```

Comment:
@smudge while attempting to clean up dependencies I ran into this strange quirk with capybara. the assertion on line 172 fails without the `sleep 1` because the page contents are actually some how still changing?

switching `expect(page).to have_no_text('The Everyperson')` to `expect(page).not_to have_text('The Everyperson')` looks semantically equivalent (in fact rubocop automatically corrected it!) but because the page doesn't have the appropriate content yet, I think the original assertion may have been incorrect?

Anyway, this is the one blocker to get the only https_github reference out of this repo. I thought it would be as simple as removing the one line in the Gemfile but the tests were failing so I'm not sure if things were always like that?

Reply:
In the versions of capybara I'm used to using, `.to have_no_text` should return immediately, while `.not_to have_text` will wait for have_text to time out waiting for the text (or return early if it finds it, and fail).

So, yeah, the assertion was wrong, but IMO rubocop's suggestion is also wrong, because it inflates test times by a lot (at least, historically). I'm not finding anything about this having changed in newer versions... 🤔

Reply:
The best practice here is to make a positive assertion (wait for something else on the page to load) before making the negative assertion.

Reply:
do you know what I could do to make that happen? I tried stuff like this `have_css('#session_variant')` which from a cursory google search shows that would cause the process to block til the page loads (which is strange since I would have expected `have_text` to do the same?)

Reply:
Alright I'm going to call it @smudge. I think part of the complications are it's not clear (to me at least) which methods wait, which ones don't, which ones check for "visibility", etc. Dumping the html before and after `sleep` calls shows it's clearly changing but the only thing actually changing is the attribute of two elements (diff attached) which afaik no Capybara API is designed to wait for.

What do you think we should do from here?

<img width="948" alt="Screen Shot 2023-02-23 at 7 28 50 PM" src="https://user-images.githubusercontent.com/47008591/221062216-c034d559-5ed3-49e0-bd57-fb3ec208bd73.png">

Reply:
👋 Hi -- I've been swamped today, so I haven't had a chance to circle back.

I think the issue here might be that "The Everyperson" appears elsewhere on the page (way at the top, above the table), so even if you write the test correctly it is going to fail. Maybe I'm wrong though -- I haven't booted up the code to test it out. Would be happy to pair on this next week to see if we can get it to a resolution 🙂 

In the meantime, I think it would also be acceptable to revert some of these linter-driven changes, and run `rubucop --auto-gen-config` instead. Or if the issue is that the existing build is _already_ failing, we could get your Gemfile change merged and follow-on with the changes necessary to getting the build fixed.